compiler/vmgen.nim

```diff
@@ -1524,8 +1525,12 @@ proc setSlot(c: PCtx; v: PSym) =
     v.position = getFreeRegister(c, if v.kind == skLet: slotFixedLet else: slotFixedVar, start = 1)
 
 proc cannotEval(c: PCtx; n: PNode) {.noinline.} =
-  globalError(c.config, n.info, "cannot evaluate at compile time: " &
+  if c.config.cmd == cmdCheck:
+    localError(c.config, n.info, "cannot evaluate at compile time: " & 
```

Comment:
Why? It's very dangerous to do this. You should at least turn `cannotEval` into a `template` then and use `return` after the `localError`.